crates/stellar-scaffold-cli/src/commands/generate/contract/mod.rs

```diff
@@ -112,14 +77,138 @@ impl Cmd {
 
+        // Get the latest release tag we're using
+        let Release { tag_name } = Self::fetch_latest_release().await?;
+
+        // Read and update workspace Cargo.toml
+        let workspace_cargo_path = Path::new("Cargo.toml");
+        if workspace_cargo_path.exists() {
+            Self::update_workspace_dependencies(
+                workspace_cargo_path,
+                &example_source_path,
+                &tag_name,
+            )?;
+        } else {
+            eprintln!("⚠️  Warning: No workspace Cargo.toml found in current directory.");
+            eprintln!("   You'll need to manually add required dependencies to your workspace.");
+        }
+
         eprintln!("✅ Successfully downloaded example '{example_name}' to {dest_path}");
         Ok(())
     }
 
+    fn update_workspace_dependencies(
+        workspace_path: &Path,
+        example_path: &Path,
+        tag: &str,
+    ) -> Result<(), Error> {
+        let example_cargo_content = fs::read_to_string(example_path.join("Cargo.toml"))?;
+        let deps = Self::extract_stellar_dependencies(&example_cargo_content)?;
+        if deps.is_empty() {
+            return Ok(());
+        }
+
+        // Parse the workspace Cargo.toml
+        let mut manifest = cargo_toml::Manifest::from_path(workspace_path)?;
+
+        // Ensure workspace.dependencies exists
+        if manifest.workspace.is_none() {
+            // Create a minimal workspace with just what we need
+            let workspace_toml = r"
+[workspace]
+members = []
+
+[workspace.dependencies]
+";
+            let workspace: cargo_toml::Workspace<toml::Value> = toml::from_str(workspace_toml)?;
+            manifest.workspace = Some(workspace);
+        }
+        let workspace = manifest.workspace.as_mut().unwrap();
+
+        let mut workspace_deps = workspace.dependencies.clone();
+
+        let mut added_deps = Vec::new();
+        let mut updated_deps = Vec::new();
+
+        for dep in deps {
+            let git_dep = cargo_toml::DependencyDetail {
+                git: Some("https://github.com/OpenZeppelin/stellar-contracts".to_string()),
+                tag: Some(tag.to_string()),
+                ..Default::default()
+            };
+
+            if let Some(existing_dep) = workspace_deps.clone().get(&dep) {
+                // Check if we need to update the tag
+                if let cargo_toml::Dependency::Detailed(detail) = existing_dep {
+                    if let Some(existing_tag) = &detail.tag {
+                        if existing_tag != tag {
+                            workspace_deps.insert(
+                                dep.clone(),
+                                cargo_toml::Dependency::Detailed(Box::new(git_dep)),
+                            );
+                            updated_deps.push((dep, existing_tag.clone()));
+                        }
+                    }
+                }
+            } else {
```

Comment:
Since we are using the newest version of rust this should work.
```suggestion
            if let Some(existing_dep) = workspace_deps.clone().get(&dep) {
                // Check if we need to update the tag
                if let Detailed(DependencyDetail { tag: Some(existing_tag), ..}) = &existing_dep && existing_tag != tag {
                      workspace_deps.insert(
                          dep.clone(),
                          cargo_toml::Dependency::Detailed(Box::new(git_dep)),
                      );
                      updated_deps.push((dep, existing_tag.clone()));
                }
            } else {
```